c++/src/kj/debug-test.c++

```diff
@@ -544,8 +558,48 @@ KJ_TEST("magic assert stringification signal handler") {
   auto expected = str(condition);
   auto buf = kj::heapArray<char>(200);
   auto actual = strPreallocated<StringifyContext::SIGNAL_HANDLER>(buf, condition);
-  KJ_REQUIRE(actual == expected);
-  KJ_REQUIRE("4294967295 == 4294967295" == actual);
+  KJ_EXPECT(actual == expected);
+  KJ_EXPECT("4294967295 == 4294967295" == actual);
+
+#if __linux__ && !__aarch64__
+  // Double-check that this works within a real signal handler. The intention is that this test runs
+  // under TSAN to have compiler-generated instrumentation verify that assumptions hold.
+  // For some reason the signal handler never ran on Arm64 builds so disabling it for now as it's
+  // not worth figuring out for what this is testing.
+  memset(buf.begin(), 0, buf.size());
+
+  static sig_atomic_t ranSignalHandler = 0;
+
+  struct sigaction action{};
+  action.sa_flags = SA_SIGINFO;
+  action.sa_sigaction = [](int, siginfo_t* info, void*) {
+    static constexpr Duration a = INT64_MIN * NANOSECONDS;
+    static constexpr Duration b = INT64_MIN * NANOSECONDS;
+    static const auto condition = _::MAGIC_ASSERT << a == b;
+    ranSignalHandler = 1;
+    auto buf = reinterpret_cast<kj::Array<char>*>(info->si_value.sival_ptr);
+    strPreallocated<StringifyContext::SIGNAL_HANDLER>(*buf, condition);
+  };
+  KJ_SYSCALL(sigaction(SIGRTMIN, &action, nullptr));
+
+  siginfo_t info{};
+  info.si_code = SI_QUEUE;
+  info.si_pid = getpid();
+  info.si_uid = getegid();
+  info.si_value.sival_ptr = &buf;
+  KJ_SYSCALL(syscall(SYS_rt_tgsigqueueinfo, getpid(), gettid(), SIGRTMIN, &info));
```

Comment:
Why the very complicated way of queuing a signal instead of just `raise(SIGRTMIN)`?

Reply:
I was using it to supply the output buffer. I guess I can just switch it to use a global instead.